pytest/lib/bridge.py

```diff
@@ -0,0 +1,236 @@
+        pass
+
+class Cleanable(object):
+
+    def __init__(self, config):
+        self.cleaned = False
+        self.pid = multiprocessing.Value('i', 0)
+        self.config = config
+        self.bridge_dir = self.config['bridge_dir']
+        self.config_dir = self.config['config_dir']
+        self.stdout = None
+        self.stderr = None
+        atexit.register(atexit_cleanup, self)
+
+    def kill(self):
+        if self.pid.value != 0:
+            os.kill(self.pid.value, signal.SIGKILL)
+            self.pid.value = 0
+    
+    def cleanup(self):
+        if self.cleaned:
+            return
+
+        try:
+            if self.stdout:
+                self.stdout.close()
+            if self.stderr:
+                self.stderr.close()
+            self.kill()
+        except:
+            print("Kill %s failed on cleanup!" % (self.__class__.__name__))
+            traceback.print_exc()
+            print("\n\n")
+
+    def restart(self):
+        assert not self.cleaned
+        self.cleanup()
+        assert self.pid.value == 0
+        self.start()
+        assert self.pid.value != 0
+        self.cleaned = False
+
+
+class GanacheNode(Cleanable):
+
+    def __init__(self, config):
+        super(GanacheNode, self).__init__(config)
+        ganache_bin = os.path.join(self.bridge_dir, self.config['ganache_bin'])
+        if not os.path.exists(ganache_bin):
+            self.build()
+
+    def build(self):
+        ganache_dir = os.path.join(self.bridge_dir, self.config['ganache_dir'])
+        assert subprocess.check_output(['yarn'], cwd=ganache_dir).decode('ascii').strip().split('\n')[-1].strip().split(' ')[0] == 'Done'
+
+    def start(self):
+        ganache_bin = os.path.join(self.bridge_dir, self.config['ganache_bin'])
+        self.stdout = open(os.path.join(self.config_dir, 'logs/ganache/out.log'), 'w')
+        self.stderr = open(os.path.join(self.config_dir, 'logs/ganache/err.log'), 'w')
+        # TODO use blockTime
+        # TODO set params
+        self.pid.value = subprocess.Popen([ganache_bin,'--port','9545','--blockTime','12','--gasLimit','10000000','--account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501200,10000000000000000000000000000"','--account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501201,10000000000000000000000000000"','--account="0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501202,10000000000000000000000000000"'], stdout=self.stdout, stderr=self.stderr).pid
+        assert self.pid.value != 0
+
+
+class Near2EthBlockRelay(Cleanable):
+
+    def __init__(self, config):
+        super(Near2EthBlockRelay, self).__init__(config)
+
+    def start(self, eth_master_secret_key='0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501201'):
+        self.stdout = open(os.path.join(self.config_dir, 'logs/near2eth-relay/out.log'), 'a')
+        self.stderr = open(os.path.join(self.config_dir, 'logs/near2eth-relay/err.log'), 'a')
```

Comment:
Not sure if it is done already, but it is a good idea to expose this logs in nayduck.

Reply:
Thanks to @utka, they are already available. Example of run: http://nayduck.eastus.cloudapp.azure.com:3000/#/run/776